libbeat/processors/actions/decode_csv_field_test.go

```diff
@@ -0,0 +1,204 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package actions
+
+import (
+	"bytes"
+	"testing"
+	"unicode"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDecodeCSVField(t *testing.T) {
+	tests := map[string]struct {
+		config   string
+		input    beat.Event
+		expected beat.Event
+		fail     bool
+	}{
+		"default target": {
+			config: `
+				field: message
+				`,
+			input: beat.Event{
+				Fields: common.MapStr{
+					"message": "17,192.168.33.1,8.8.8.8",
+				},
+			},
+			expected: beat.Event{
+				Fields: common.MapStr{
+					"message": "17,192.168.33.1,8.8.8.8",
+					"csv":     []string{"17", "192.168.33.1", "8.8.8.8"},
+				},
+			},
+		},
+		"alternative target": {
+			config: `
+				field: message
+				target: my.field
+				`,
+			input: beat.Event{
+				Fields: common.MapStr{
+					"message": "17,192.168.33.1,8.8.8.8",
+				},
+			},
+			expected: beat.Event{
+				Fields: common.MapStr{
+					"message":  "17,192.168.33.1,8.8.8.8",
+					"my.field": []string{"17", "192.168.33.1", "8.8.8.8"},
+				},
+			},
+		},
+		"no field set": {
+			config: ``,
+			fail:   true,
+		},
+		"non existing field": {
+			config: `
+				field: my_field`,
+			fail: true,
+		},
+		"ignore missing": {
+			config: `
+				field: my_field
+				ignore_missing: true
+				`,
+		},
+		"overwrite keys failure": {
+			config: `
+				field: message
+				target: existing_field
+				`,
+			input: beat.Event{
+				Fields: common.MapStr{
+					"message":        `"hello ""world"""`,
+					"existing_field": 42,
+				},
+			},
+			fail: true,
+		},
+
+		"overwrite keys": {
+			config: `
```

Comment:
Personally I would have used `common.MustNewConfigFrom(common.MapStr{...})` to directly define a common.Config and avoid the `unindent` helper.